.npmrc

```diff
@@ -1,2 +1 @@
 save-exact = true
```

Comment:
removing this one may raise some eyebrows. But I don't really think we need it anymore. Automatic updating of the package lock seems like it's been pretty solid since 5.8.

Reply:
What I think `save-exact` does: save package version in `package.json` as `react: 16.4.0` rather than `react: 16.4.0^`. And prevent accidental upgrade when I run `npm i` on a clean directory or regenerate the lock with `npm run update-deps`.

Is that really what it does? If `save-exact` is not there, how do we prevent the "everything gets upgraded to latest version compliant with the `^` op" scenario?

Reply:
ah, no, i meant removing `package-lock=false` on the next line. I just couldn't find a way to leave a comment on that line in the GH UI.

Reply:
and `save-exact` is exactly what we want.

Reply:
> Automatic updating of the package lock seems like it's been pretty solid since 5.8.

The point of `package-lock=false` is to force `npm` to use `npm-shrinkwrap.json` instead of `package-lock.json`. We want to keep it that way, unless `package-lock.json` is generated with a different algorithm than `npm-shrinkwrap.json` in `npm@6`.

Reply:
if there's a npm-shrinkwrap.json present, npm will update that instead of package-lock.json. At least that's what I've found in testing. Are you seeing something different?

Reply:
I totally misinterpreted [what this option does](https://docs.npmjs.com/misc/config#package-lock). My apologies, please ignore me :/